Colour documentation RSC-143

glenhunter opened the pull request:
https://issues.sonatype.org/browse/RSC-143

<img width="1189" alt="Screen Shot 2021-02-18 at 1 34 25 PM" src="https://user-images.githubusercontent.com/1991443/108404617-13023900-71ee-11eb-9900-f8ffb5bf391e.png">
<img width="1179" alt="Screen Shot 2021-02-18 at 1 34 33 PM" src="https://user-images.githubusercontent.com/1991443/108404622-14336600-71ee-11eb-9ec2-442f998409c6.png">

rpokorny commented:
There's also `$nx-white`, should that be documented?  It is still used but I guess didn't make it into any of the swatches due to its lack of hue

rpokorny commented on gallery/package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@sonatype/react-shared-components-gallery",
-  "version": "3.7.0",
+  "version": "3.8.0",
```
This is just a documentation update so should be a patch version

glenhunter replied:
> This is just a documentation update so should be a patch version

De-bumped: https://github.com/sonatype/sonatype-react-shared-components/pull/313/commits/d03f60260740793e3b6c4fd01e41d55ead98257f

rpokorny commented on gallery/src/guidelines/ColorPalette/ColorPalettePage.tsx:
```diff
@@ -0,0 +1,57 @@
+  <>
+    <GalleryDescriptionTile>
+      <p className="nx-p">
+        The design group have created a palette of colors for use in RSC. The colors are organized into swatches
```
```suggestion
        The design group has created a palette of colors for use in RSC. The colors are organized into swatches
```

rpokorny commented on gallery/src/guidelines/ColorPalette/ColorPalettePage.tsx:
```diff
@@ -0,0 +1,57 @@
+        <p className="nx-p">
+          Where a color is used commonly across multiple components a custom variable is created. For example many
+          components with a disabled state have the same background color so we have created a color variable:
+          {' '}<code className="nx-code">$nx-disabled-background: $nx-grey-200;</code>. If the disabled background color
```
I'm not sure we should mention this unless we are going to document what additional variables are supported.  Especially since `_nx-colors.scss` currently contains a mix of up-to-date semantic color variables, and deprecated color variables that aren't part of the current system and which we are theoretically working on removing.

glenhunter replied:
> I'm not sure we should mention this unless we are going to document what additional variables are supported. Especially since `_nx-colors.scss` currently contains a mix of up-to-date semantic color variables, and deprecated color variables that aren't part of the current system and which we are theoretically working on removing.

My goal was to explain why sometimes we use `$nx-disabled-background` instead of `$nx-grey-200` (for example), I agree that we don't want to list all of the named variables, but I do think that the explanation has value.

rpokorny replied:
> My goal was to explain why sometimes we use $nx-disabled-background instead of $nx-grey-200 (for example), I agree that we don't want to list all of the named variables, but I do think that the explanation has value.

I'm worried that it prompts users to use colors from `_nx-colors.scss` and that's not currently a safe idea.

glenhunter replied:
> > My goal was to explain why sometimes we use $nx-disabled-background instead of $nx-grey-200 (for example), I agree that we don't want to list all of the named variables, but I do think that the explanation has value.
> 
> I'm worried that it prompts users to use colors from `_nx-colors.scss` and that's not currently a safe idea.

The problem is that there are many variables in `_color.scss` that we _do_ want people to use rather than a `nx-color-###` variable. I think that this isn't a problem with this documentation but rather a problem with `_colors.scss`. I know that we had a discussion a little while ago about moving some of those variables down to the bottom of the file into a "deprecated" section. Maybe we should up the priority of that task.

rpokorny replied:
> The problem is that there are many variables in _color.scss that we do want people to use rather than a nx-color-### variable. I think that this isn't a problem with this documentation but rather a problem with _colors.scss. I know that we had a discussion a little while ago about moving some of those variables down to the bottom of the file into a "deprecated" section. Maybe we should up the priority of that task.

Right.  But until we clean that up, I think we shouldn't document it in the way you have here.  We can make sure we document the desirable variables later once they are distinguishable.

glenhunter replied:
> > The problem is that there are many variables in _color.scss that we do want people to use rather than a nx-color-### variable. I think that this isn't a problem with this documentation but rather a problem with _colors.scss. I know that we had a discussion a little while ago about moving some of those variables down to the bottom of the file into a "deprecated" section. Maybe we should up the priority of that task.
> 
> Right. But until we clean that up, I think we shouldn't document it in the way you have here. We can make sure we document the desirable variables later once they are distinguishable.

Removed here and I created: https://issues.sonatype.org/browse/RSC-496

rpokorny commented on gallery/src/guidelines/ColorPalette/Swatcher.scss:
```diff
@@ -0,0 +1,43 @@
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+ @import '~@sonatype/react-shared-components/scss-shared/nx-colors';
```
Redundant; `nx-variables` imports this

glenhunter replied:
> Redundant; `nx-variables` imports this

Ooops: https://github.com/sonatype/sonatype-react-shared-components/pull/313/commits/2add318c9ea7af6b13b90cacc20d0452876b0014

rpokorny commented on gallery/src/guidelines/ColorPalette/ColorPalettePage.tsx:
```diff
@@ -0,0 +1,57 @@
+          to replace colors in the future if the palette changes.
+        </p>
+      </section>
+    </GalleryDescriptionTile>
```
Do we want to describe the preferred way of importing the color variables?

glenhunter replied:
> Do we want to describe the preferred way of importing the color variables?

Good idea: https://github.com/sonatype/sonatype-react-shared-components/pull/313/commits/b40c16b8122633991ea811a49a9bd48050d5abfd

rpokorny commented on gallery/src/guidelines/ColorPalette/Swatcher.scss:
```diff
@@ -0,0 +1,43 @@
+ @import '~@sonatype/react-shared-components/scss-shared/nx-variables';
+ @import '~@sonatype/react-shared-components/scss-shared/nx-text-helpers';
+
+.gallery-swatches {
```
I wonder if these would be good candidates to make as `.nx-card`s

glenhunter replied:
> I wonder if these would be good candidates to make as `.nx-card`s

You know I did think about that, I didn't just because the `NxCard` PR is still in review. We could hold on this until it's merged.

glenhunter replied:
> > I wonder if these would be good candidates to make as `.nx-card`s
> 
> You know I did think about that, I didn't just because the `NxCard` PR is still in review. We could hold on this until it's merged.

`.nx-card` implementation. I tweaked the `max-width` a bit so that the longer variables didn't wrap.

rpokorny commented on gallery/src/guidelines/ColorPalette/Swatcher.scss:
```diff
@@ -0,0 +1,43 @@
+}
+
+.gallery-swatch__hex {
+  flex-basis: 7ch;
```
What an odd choice of unit.  Why?

rpokorny replied:
> What an odd choice of unit. Why?

Oh I guess because a full width color code (assuming no transparency) is 7 characters wide? Keep in mind that's only precise in a fixed-width font

glenhunter replied:
> > What an odd choice of unit. Why?
> 
> Oh I guess because a full width color code (assuming no transparency) is 7 characters wide? Keep in mind that's only precise in a fixed-width font

yeah, seemed to work well here though

glenhunter commented:
> There's also `$nx-white`, should that be documented? It is still used but I guess didn't make it into any of the swatches due to its lack of hue

I added black and white here: https://github.com/sonatype/sonatype-react-shared-components/pull/313/commits/d3882e547f5086a0d86f06cc4cead2c0b3de18c5

When I have the cards set up we can see how it works and whether we want to keep them. I suspect we won't want to

rpokorny commented:
> I added black and white here: d3882e5
>
> When I have the cards set up we can see how it works and whether we want to keep them. I suspect we won't want to

You know those aren't getting included in the gallery page, right?

rpokorny commented on lib/src/scss-shared/_nx-color-swatches.scss:
```diff
@@ -150,3 +150,9 @@ $nx-indigo-300: #bbc5f6;
+$nx-black: #000;
+
+// White
+$nx-white: #fff;
```
`$nx-white` and `$nx-black` were already present in `_nx-colors.scss`, so if you add them here you should remove them there.  I would think it wouldn't compile with them present in both places, actually.

glenhunter replied:
> `$nx-white` and `$nx-black` were already present in `_nx-colors.scss`, so if you add them here you should remove them there. I would think it wouldn't compile with them present in both places, actually.

Sure it's not problem it just overrides the value of the variable based on load order. Regardless I removed them, the more I thought about it the less value that I saw in having just black and white. https://github.com/sonatype/sonatype-react-shared-components/pull/313/commits/7d76da37609656f5400fcb600c1dd7586dd77efe

glenhunter commented:
> > I added black and white here: [d3882e5](https://github.com/sonatype/sonatype-react-shared-components/commit/d3882e547f5086a0d86f06cc4cead2c0b3de18c5)
> > When I have the cards set up we can see how it works and whether we want to keep them. I suspect we won't want to
> 
> You know those aren't getting included in the gallery page, right?

hah, I removed them: https://github.com/sonatype/sonatype-react-shared-components/pull/313/commits/7d76da37609656f5400fcb600c1dd7586dd77efe

rpokorny commented on gallery/package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@sonatype/react-shared-components-gallery",
-  "version": "4.1.0",
+  "version": "4.1.10",
```
.10??

glenhunter replied:
> .10??

Weird: https://github.com/sonatype/sonatype-react-shared-components/pull/313/commits/ac12d51610a479191198d08428b95d92cf96c7c9

rpokorny commented:
Minor: I'm finding it a little strange that the cards aren't all the same width.  Probably not worth worrying about though. 
![image](https://user-images.githubusercontent.com/3630091/108855407-f895da00-75b6-11eb-8c30-94b38a8f305d.png)

rpokorny commented:
> > > I added black and white here: d3882e5
> > > When I have the cards set up we can see how it works and whether we want to keep them. I suspect we won't want to
> > 
> > You know those aren't getting included in the gallery page, right?
>
> hah, I removed them: 7d76da3

Ok, well that brings us back to the beginning of the discussion: https://github.com/sonatype/sonatype-react-shared-components/pull/313#issuecomment-782181260

rpokorny commented on gallery/src/guidelines/ColorPalette/ColorPalettePage.tsx:
```diff
@@ -0,0 +1,47 @@
+          to replace colors in the future if the palette changes.
+        </p>
+        <p className="nx-p">
+          To access to the color variables you'll need to load
```
```suggestion
          To access the color variables you'll need to load
```

glenhunter commented:
> > > > I added black and white here: [d3882e5](https://github.com/sonatype/sonatype-react-shared-components/commit/d3882e547f5086a0d86f06cc4cead2c0b3de18c5)
> > > > When I have the cards set up we can see how it works and whether we want to keep them. I suspect we won't want to
> > > 
> > > 
> > > You know those aren't getting included in the gallery page, right?
> > 
> > 
> > hah, I removed them: [7d76da3](https://github.com/sonatype/sonatype-react-shared-components/commit/7d76da37609656f5400fcb600c1dd7586dd77efe)
> 
> Ok, well that brings us back to the beginning of the discussion: [#313 (comment)](https://github.com/sonatype/sonatype-react-shared-components/pull/313#issuecomment-782181260)

I don't think that we need to show black or white here.

glenhunter commented:
> Minor: I'm finding it a little strange that the cards aren't all the same width. Probably not worth worrying about though.

Ah, just adding `flex-grow:1` to the cards made it nearly perfect while still respecting the `max-width` and working responsively.
